### PR TITLE
DRILL-3149: TextReader should support multibyte line delimiters

### PR DESCRIPTION
**Pull request by arina-ielchiieva**

### CODE REVIEW

**Comment by parthchandra** [comment]
Looks Good. Can you add another test case where a field contains a some of the initial bytes of the multibyte line delimiter.

**Comment by arina-ielchiieva** [comment]
@parthchandra 
I have added / updated several unit test with second commit. Could you please take a look?

File: exec/java-exec/src/test/java/org/apache/drill/TestSelectWithOption.java [hunk 1]
```diff
@@ -89,30 +88,51 @@ public void testTabFieldDelimiter() throws Exception {
...
+        "b|1",
+        "b|2");
 
     testWithResult(format("select columns from table(%s(type => 'TeXT', lineDelimiter => '|'))", tableName),
-        listOf("\"b\""),
-        listOf("\"0\"", "\"b\""),
-        listOf("\"1\"", "\"b\""),
-        listOf("\"2\"")
-      );
+        listOf("b"),
```
Comment (parthchandra) [inline exec/java-exec/src/test/java/org/apache/drill/TestSelectWithOption.java]: The original test has the data with quotes around it. Why change it? That is a legitimate case that should be handled correctly.

File: exec/java-exec/src/test/java/org/apache/drill/TestSelectWithOption.java [hunk 2]
```diff
@@ -89,30 +88,51 @@ public void testTabFieldDelimiter() throws Exception {
...
-        listOf("1,a"),
-        listOf("2,b"));
+        listOf("1, a"),
+        listOf("2, b"));
+  }
+
+  @Test
+  public void testMultiByteLineDelimiter() throws Exception {
+    String tableName = genCSVTable("testMultiByteLineDelimiter",
+        "1abc2abc3abc");
```
Comment (parthchandra) [inline exec/java-exec/src/test/java/org/apache/drill/TestSelectWithOption.java]: I was expecting a test where the data has part of a multibyte delimiter in it.
Data -  ab1abc2abc3abc
delimiter - abc
Should output - ab1, 2, 3

Comment (arina-ielchiieva) [inline exec/java-exec/src/test/java/org/apache/drill/TestSelectWithOption.java]: @parthchandra 
Sorry for misunderstanding, added third commit with new test - testDataWithPartOfMultiByteLineDelimiter.

**Comment by parthchandra** [comment]
+1 looks good. Can you squash and rebase ?

**Comment by arina-ielchiieva** [comment]
Done.

**Comment by arina-ielchiieva** [comment]
Changed merged into master with commit id - 223507b76ff6c2227e667ae4a53f743c92edd295

**Pull request closed without merging**